src/Features/CSharp/Portable/EditAndContinue/CSharpEditAndContinueAnalyzer.cs

```diff
@@ -3019,10 +3048,43 @@ private static bool IsSimpleAwaitAssignment(SyntaxNode node, SyntaxNode awaitExp
             SyntaxNode newActiveStatement,
             bool isLeaf)
         {
+            ReportRudeEditsForUnsupportedCSharp7EnC(diagnostics, match, oldActiveStatement, newActiveStatement);
             ReportRudeEditsForAncestorsDeclaringInterStatementTemps(diagnostics, match, oldActiveStatement, newActiveStatement, isLeaf);
             ReportRudeEditsForCheckedStatements(diagnostics, oldActiveStatement, newActiveStatement, isLeaf);
         }
 
+        private void ReportRudeEditsForUnsupportedCSharp7EnC(List<RudeEditDiagnostic> diagnostics, Match<SyntaxNode> match, SyntaxNode oldActiveStatement, SyntaxNode newActiveStatement)
+        {
+            SyntaxNode foundCSharp7Syntax = match.NewRoot.DescendantNodesAndSelf().FirstOrDefault(n => IsUnsupportedCSharp7EnCNode(n));
+            if (foundCSharp7Syntax != null)
+            {
+                AddRudeDiagnostic(diagnostics, oldActiveStatement, foundCSharp7Syntax, newActiveStatement.Span);
+            }
+        }
+
+        private static bool IsUnsupportedCSharp7EnCNode(SyntaxNode n)
+        {
+            switch (n.Kind())
+            {
+                case SyntaxKind.IsPatternExpression:
+                case SyntaxKind.DeconstructionDeclarationStatement:
```

Comment:
Sure. That scenario was already caught by the tuples check, but this will make it safer to incrementally enable scenarios.